### PR TITLE
Configure via Constructor; Use new Base class

### LINKED ISSUE #17: Needs description on usage/configuration/etc
README needs a serious overhaul describing what it's doing.

Missing:
 - intent
 - usage
 - configuration

### LINKED ISSUE #20: Configuration in code not just environment
Currently you can change the host and token via the environment variables: `K8S_HOST` and `K8S_TOKEN`

You should be able to do the same during instantiation.

### PR DESCRIPTION
**Pull request by nkatzman**
Resolves #20 
Resolves #17 

This PR changes the configuration of the token and host to be done via the constructor of the object

It also adds in the new screwdriver-executor-base (https://github.com/screwdriver-cd/executor-base) class that enables input validation of executor functions

Also added in a README

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,30 +1,31 @@
...
-const k8sCluster = process.env.K8S_HOST || 'kubernetes';
-const jobsUrl = `https://${k8sCluster}/apis/batch/v1/namespaces/default/jobs`;
-const podsUrl = `https://${k8sCluster}/api/v1/namespaces/default/pods`;
 
 class K8sExecutor extends Executor {
 
     /**
      * Constructor
      * @method constructor
+     * @param  {Object}
```
Comment (tkyi) [inline index.js]: Add comments for `options`

File: package.json [hunk 2]
```diff
@@ -44,7 +44,7 @@
     "hoek": "^4.0.1",
     "js-yaml": "^3.6.1",
     "request": "^2.72.0",
-    "screwdriver-executor-base": "0.0.4",
+    "screwdriver-executor-base": "1.0.0",
```
Comment (stjohnjohnson) [inline package.json]: `^` plz

**Comment by d2lam** [comment]
LGTM 👍

**Comment by stjohnjohnson** [comment]
Woot 👍 
you_want_pancakes_i_ll_give_you_some_fuckin__pancakes_

**Pull request merged**